lab_3_generate_by_ngrams/main.py

```diff
@@ -143,6 +224,22 @@ def _decode(self, corpus: tuple[int, ...]) -> Optional[tuple[str, ...]]:
         In case of corrupt input arguments, None is returned.
         In case any of methods used return None, None is returned.
         """
+        if not isinstance(corpus, tuple) or len(corpus) == 0:
```

Comment:
Empty sequences map to False in Python, so there is no need to check it with `len` and `==`